dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -1877,12 +1877,8 @@ private String makeNodeObject()
         sb.append(" \"node\" : {\n ");
 
         DataPartition.FileType fileType = this.dataPart.getFileType();
-        boolean needToSpecifyFileType = fileType.typeCanBeDeduced() == false;
-        if (needToSpecifyFileType)
-        {
-            sb.append("\"kind\" : \"");
-            sb.append(fileType.toString() + "read\",\n");
-        }
+        sb.append("\"kind\" : \"");
+        sb.append(fileType.toString() + "read\",\n");
```

Comment:
is filetype guaranteed to be valid?

Reply:
That is a good question. We are given the file type by ESP so it should always be valid, and if it is incorrect would indicate a bug elsewhere.